### PR TITLE
nvme-cli: 2.7.1 -> 2.8

### PR DESCRIPTION
**Pull request by felixsinger**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by felixsinger** [comment]
@GrahamcOfBorg build libnvme nvme-cli

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 1]
```diff
@@ -60,7 +60,8 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   # mocked ioctl conflicts with the musl one: https://github.com/NixOS/nixpkgs/pull/263768#issuecomment-1782877974
-  doCheck = !stdenv.hostPlatform.isMusl;
+  # tests/sysfs is failing, so disable completely.
+  doCheck = false;
```
Comment (fogti) [inline pkgs/os-specific/linux/libnvme/default.nix:64]: wouldn't it make more sense to just disable `tests/sysfs` specifically?
https://github.com/linux-nvme/libnvme/blob/ea5600c843080db45590b5997176cc13d7a290bb/test/meson.build#L102

Comment (felixsinger) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: Yeah, I still was about to figure that out, if there is any possibility to do that. That's why I've marked the pull request as draft. Sorry for the confusion.

Anyway, do you have an idea on how to do that properly?

Comment (fogti) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: what does applying a patch which just comments out the linked subdir block do?

Comment (felixsinger) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: I thought about that as well. It probably does the job, but that way the patch needs to be always carried. That doesn't feel like a long-term solution.

Comment (Mic92) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: Does not sound that complex of a patch.

Comment (fogti) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: it's like a one-line change, it could even be done with sed/substituteInPlace, probably

Comment (felixsinger) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: I like the idea to use sed. Will so that later!

Comment (mmlb) [reply pkgs/os-specific/linux/libnvme/default.nix:64]: I'ved opened up #309932 with newer updates and fix for the sysfs shebang, ptal there?

**Comment by felixsinger** [comment]
Superseded by #309932.

**Pull request closed without merging**